multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -208,6 +215,64 @@ def is_imagepath_column(
         return False
 
 
+def is_imagebytearray_column(
+    data: pd.Series,
+    col_name: str,
+    sample_n: Optional[int] = 500,
+) -> bool:
+    """
+    Identify if a column contains valid image bytearray.
+
+    Parameters
+    ----------
+    data
+        One column of a multimodal pd.DataFrame for training.
+    col_name
+        Name of column.
+    sample_n
+        Number of sample images to open for sanity check.
+
+    Returns
+    -------
+    Whether the column is an image-bytearray column.
+    """
+    sample_num = min(len(data), sample_n)
+    data = data.sample(n=sample_num, random_state=0)
+    data = data.tolist()
+    failure_count = 0
+    for image_bytearrays in data:
+        success = False
+        if not isinstance(image_bytearrays, list):
+            image_bytearrays = [image_bytearrays]
+
+        for img_bytearray in image_bytearrays:
+            try:
+                with PIL.Image.open(BytesIO(img_bytearray)) as img:
```

Comment:
What if the image bytes are encoded by base64?

Reply:
Can we enforce that the type passed into the predictor to be the original `bytearray`? In the cloud serve scripts, I see that currently we would: 1. decode the encoded bytearray in a predefined way, and 2. save to disk to get the path to pass into the predictor. With the new feature, we can save step 2 by assuming no encoding on bytearray. If we are to expect encoded bytearrays in the dataframe, we would probably need to accept a callback decoding function provided by the user. What do you think?

Reply:
base64 is used in encoding images in some cases. I think we'd better support users to provide base64 bytes in dataframe.

Reply:
`base64` is fundamentally a different format than `bytearray`, since it is a text encoding rather than a binary encoding. We could consider adding a `IMAGE_BASE64` column type, or we could try to make a more generic `IMAGE_BINARY` type that also infers the encoding. But I think this is another feature request separate from the current issue.

Reply:
Since base64 requires additional decoding, we may define another type `IMAGE_BASE64`. We can leave this to another PR.